### PR TITLE
Get more verbose logging for failures

### PR DESCRIPTION
**Pull request by matthew-andrews**
To help diagnosing future problems with article failures.
@i-like-robots

### CODE REVIEW

File: server/lib/fetch-capi-json.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-				throw new Error(`Bad response from ${endpoint}`);
+		.then((response) => {
+			if (response.ok) {
+				return response.json();
 			}
-
-			return response.json();
+			return response.text()
+				.then((text) => {
+					return Promise.reject({ error: text, status: response.status });
```
Comment (matthew-andrews) [inline server/lib/fetch-capi-json.js]: critically, passing the text that CAPI **did** return with along with the status back up through the logs.

File: server/models/article-v3.js [hunk 2]
```diff
@@ -14,14 +14,14 @@ const sillyPrefixRegex = /http:\/\/(?:www|api)\.ft\.com\/things?\//;
...
-			logger.warn(`event=CAPI_V1_FETCH_FAIL region=${REGION} index=v3 uuid=${articleId} error='${error.toString()}'`);
+			logger.warn({ event: 'CAPI_V1_FETCH_FAIL', region: REGION, index: 'v3', uuid: articleId, error: error.error, status: error.status });
 		});
 }
 
 function fetchCapiV2(articleId) {
 	return fetchCapiJson(`http://api.ft.com/enrichedcontent/${articleId}`)
 		.catch(function(error) {
-			logger.warn(`event=CAPI_V2_FETCH_FAIL region=${REGION} index=v3 uuid=${articleId} error='${error.toString()}'`);
+			logger.warn({ event: 'CAPI_V2_FETCH_FAIL', region: REGION, index: 'v3', uuid: articleId, error: error.error, status: error.status });
```
Comment (i-like-robots) [inline server/models/article-v3.js]: The index property is probably irrelevant now I guess

**Comment by i-like-robots** [comment]
:+1:

File: server/models/article-v3.js (hunk 2 above)
Comment (matthew-andrews) [inline server/models/article-v3.js]: yeah i guess

**Pull request merged**